### DISCUSSION TITLE
How to do retry for async method without using executorService?

### DISCUSSION THREAD

**Discussion by DiptiGupta** [resilience4j/resilience4j#2195] [Q&A] [answered]
I want to do retry for async method which returns CompletableFuture as shown below

public CompletableFuture<Void> saveData(A obj) {
 return dao.save(obj);
}

It worked for me Retry.decorateCompletionStage but this method needs an argument of type ScheduledExecutorService.
I don’t want to create and pass any executorService. 
Is there any other way to apply retry for saveData?

**Comment by RobWin** [comment, marked as answer]
No, there is no other way. You need to provide an ExecutorService so that Resilience4j can create a new Thread which is used to execute the retry attempt with the configured delay time. The existing thread cannot be reused.
